evaluator/builtin_time.go

```diff
@@ -601,6 +601,69 @@ func builtinYearWeek(args []types.Datum, _ context.Context) (types.Datum, error)
 	return d, nil
 }
 
+// http://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_from-unixtime
+func builtinFromUnixTime(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	unixTimeStamp, err := args[0].ToDecimal()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	// 0 <= unixTimeStamp <= INT32_MAX
+	if unixTimeStamp.IsNegative() {
+		return
+	}
+	integralPart, err := unixTimeStamp.ToInt()
+	if err != nil && err != mysql.ErrTruncated {
+		return d, errors.Trace(err)
+	}
+	if integralPart > int64(1<<31-1) {
+		return
```

Comment:
When the decimal is a float, and the fractional part is not 0, it will get a `mysql.ErrTruncated` err when we  invoke `ToInt`.
Here we use `ToInt` to get the integral part of a decimal, so we can ignore this error.